ssc/cmod_battery.cpp

```diff
@@ -812,22 +816,43 @@ battstor::battstor(var_table& vt, bool setup_model, size_t nrec, double dt_hr, c
                                             batt_vars->batt_voltage_matrix, batt_vars->batt_resistance,
                                             dt_hr);
 
-    if (batt_vars->batt_calendar_choice == lifetime_params::CALENDAR_CHOICE::MODEL) {
-        lifetime_model = new lifetime_t(batt_vars->batt_lifetime_matrix, dt_hr,
-                                        batt_vars->batt_calendar_q0, batt_vars->batt_calendar_a, batt_vars->batt_calendar_b, batt_vars->batt_calendar_c);
+    if (batt_vars->batt_life_model == 0) {
+        if (batt_vars->batt_calendar_choice == calendar_cycle_params::CALENDAR_CHOICE::MODEL) {
+            lifetime_model = new lifetime_calendar_cycle_t(batt_vars->batt_lifetime_matrix, dt_hr,
+                                                           batt_vars->batt_calendar_q0, batt_vars->batt_calendar_a, batt_vars->batt_calendar_b, batt_vars->batt_calendar_c);
+        }
+        else if (batt_vars->batt_calendar_choice == calendar_cycle_params::CALENDAR_CHOICE::TABLE) {
+            lifetime_model = new lifetime_calendar_cycle_t(batt_vars->batt_lifetime_matrix, dt_hr, batt_vars->batt_calendar_lifetime_matrix);
+        }
+        else {
+            lifetime_model = new lifetime_calendar_cycle_t(batt_vars->batt_lifetime_matrix, dt_hr);
+        }
     }
-    else if (batt_vars->batt_calendar_choice == lifetime_params::CALENDAR_CHOICE::TABLE) {
-        lifetime_model = new lifetime_t(batt_vars->batt_lifetime_matrix, dt_hr, batt_vars->batt_calendar_lifetime_matrix);
+    else if (batt_vars->batt_life_model == 1) {
+        lifetime_model = new lifetime_nmc_t(dt_hr);
```

Comment:
Do we need to throw exceptions based on the chosen voltage model if the user chooses this lifetime model? Should it be possible to run this lifetime model with a lead acid battery? (it looks like it currently is possible, at least from PySAM)

Reply:
Good point, added an error check